src/models/network.js

```diff
@@ -128,6 +128,22 @@ Network.prototype.getNetworkStatus = function() {
 	return status;
 };
 
+Network.prototype.addChannel = function(newChan) {
+	let index = this.channels.length - 1;
+
+	// Don't sort special channels in amongst channels/users.
+	if (newChan.type === Chan.Type.CHANNEL || newChan.type === Chan.Type.QUERY) {
+		for (let i = 0; i < this.channels.length; i++) {
+			if (newChan.name < this.channels[i].name) {
```

Comment:
Make sure the sort is not case-sensitive (that would also be a nice test case).

Reply:
Oh it will be case sensitive. Good catch.

Reply:
Sorted